great_expectations/experimental/datasources/filesystem_data_asset.py

```diff
@@ -49,22 +45,22 @@ def test_connection(self) -> None:
         Raises:
             TestConnectionError: If the connection test fails.
         """
-        datasource: PandasFilesystemDatasource | SparkDatasource = self.datasource
-
         success = False
-        for filepath in datasource.base_directory.iterdir():
-            if self.regex.match(filepath.name):
+        for filepath in self.datasource.base_directory.glob("**/*.*"):
```

Comment:
This change enables the case where our `regex` references a directory structure, as opposed to a filename.